client/src/context/AuthContext.tsx

```diff
@@ -0,0 +1,118 @@
+import React, {
+  FC,
+  createContext,
+  useCallback,
+  useState,
+  useMemo,
+} from 'react';
+import axios, { AxiosError } from 'axios';
+import {
+  AuthContextData,
+  User,
+  signupProps,
+  CustomErrorResponse,
+} from '../interfaces';
+
+export const AuthContext = createContext<AuthContextData>({
+  user: null,
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  login: async () => {},
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  signup: async () => {},
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  logout: async () => {},
+  errorMessage: '',
+});
+
+export const AuthProvider: FC<React.PropsWithChildren<object>> = ({
+  children,
+}) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [errorMessage, setErrorMessage] = useState<string>('');
+
+  function isAxiosError(error: unknown): error is AxiosError {
+    return (error as AxiosError).isAxiosError !== undefined;
+  }
+
+  const login = useCallback(async (username: string, password: string) => {
+    try {
+      const response = await axios.post(`/api/v1/user/login`, {
+        username,
+        password,
+      });
+
+      setUser(response.data.data.user);
+      localStorage.setItem('user', JSON.stringify(response.data.data.user));
+      window.location.href = '/home';
```

Comment:
in React there is substitute for this : 
example : 
```
import { useHistory } from 'react-router-dom';
const MyComponent = () => {
  const history = useHistory();

  const redirectToHome = () => {
    history.push('/home');
  };

  return (
    <button onClick={redirectToHome}>
      Go to Home
    </button>
  );
};
```

Reply:
or you can use 'useNavigate' hoke to navigate user to the home page 
For example 

``` const navigate = useNavigate()

const OnClick = ()=>{
navigate('/home')
}
```